### PR TITLE
fix: depend on datamodel version headers in datamodel glue

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
This will force rebuilding of the datamodel glue headers in local development copies when the data model changes version.

Depends on:
- [x] upgrade eic-shell to EDM4hep 0.9, https://eicweb.phy.anl.gov/containers/eic_container/-/merge_requests/689
- [x] roll out eic-shell with EDM4hep 0.9 to cvmfs and github actions
- [ ] ~~update minimum required EDM4hep version to 0.9~~ (compatibility is ensured by febc5f78674ab65ede01b617e950fe46564bd38b)
- [x] update EDM4eic to provide version header, https://github.com/eic/EDM4eic/pull/40,
- [x] release new EDM4eic version 2.1.0
- [x] upgrade eic-spack to provide EDM4eic version 2.1.0, https://github.com/eic/eic-spack/pull/521
- [x] upgrade eic-shell to EDM4eic 2.1, https://eicweb.phy.anl.gov/containers/eic_container/-/merge_requests/690
- [x] roll out eic-shell with EDM4eic 2.1 to cvmfs and github actions
- [ ] ~~update minimum required EDM4hep version to 2.1~~ (compatibility is ensured by febc5f78674ab65ede01b617e950fe46564bd38b )

### What kind of change does this PR introduce?
- [x] Bug fix (issue: `cmake --fresh` does not rebuild datamodel glue header when switching datamodel versions)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.

### Does this PR change default behavior?
No.

### CODE REVIEW

File: src/services/io/podio/CMakeLists.txt [hunk 1]
```diff
@@ -33,12 +33,15 @@ get_target_property(EDM4EIC_INTERFACE_INCLUDE_DIRECTORIES EDM4EIC::edm4eic INTER
...
-        MAIN_DEPENDENCY ${PROJECT_SOURCE_DIR}/make_datamodel_glue.py
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/datamodel_glue.h ${CMAKE_CURRENT_BINARY_DIR}/datamodel_includes.h
+    COMMAND python3 ${PROJECT_SOURCE_DIR}/make_datamodel_glue.py
+        WORKING_DIR=${CMAKE_CURRENT_BINARY_DIR}
+        EDM4HEP_INCLUDE_DIR=${EDM4HEP_INTERFACE_INCLUDE_DIRECTORIES}
+        EDM4EIC_INCLUDE_DIR=${EDM4EIC_INTERFACE_INCLUDE_DIRECTORIES}
+    MAIN_DEPENDENCY ${PROJECT_SOURCE_DIR}/make_datamodel_glue.py
+    DEPENDS
+        ${EDM4HEP_INTERFACE_INCLUDE_DIRECTORIES}/edm4hep/EDM4HEPVersion.h
+        ${EDM4EIC_INTERFACE_INCLUDE_DIRECTORIES}/edm4eic/EDM4EICVersion.h
```
Comment (veprbl) [inline src/services/io/podio/CMakeLists.txt:44]:
```suggestion
        ${EDM4HEP_INTERFACE_INCLUDE_DIRECTORIES}/edm4hep/EDM4hepVersion.h
        ${EDM4EIC_INTERFACE_INCLUDE_DIRECTORIES}/edm4eic/EDM4eicVersion.h
```

Comment (wdconinc) [reply src/services/io/podio/CMakeLists.txt:44]: Only as of 0.9, https://github.com/key4hep/EDM4hep/commit/1145e20b9ee238e1110f643232cabd061646582e

Comment (wdconinc) [reply src/services/io/podio/CMakeLists.txt:44]: https://eicweb.phy.anl.gov/containers/eic_container/-/merge_requests/689

File: src/services/io/podio/CMakeLists.txt [hunk 2]
```diff
@@ -33,12 +33,15 @@ get_target_property(EDM4EIC_INTERFACE_INCLUDE_DIRECTORIES EDM4EIC::edm4eic INTER
...
-        MAIN_DEPENDENCY ${PROJECT_SOURCE_DIR}/make_datamodel_glue.py
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/datamodel_glue.h ${CMAKE_CURRENT_BINARY_DIR}/datamodel_includes.h
+    COMMAND python3 ${PROJECT_SOURCE_DIR}/make_datamodel_glue.py
+        WORKING_DIR=${CMAKE_CURRENT_BINARY_DIR}
+        EDM4HEP_INCLUDE_DIR=${EDM4HEP_INTERFACE_INCLUDE_DIRECTORIES}
+        EDM4EIC_INCLUDE_DIR=${EDM4EIC_INTERFACE_INCLUDE_DIRECTORIES}
+    MAIN_DEPENDENCY ${PROJECT_SOURCE_DIR}/make_datamodel_glue.py
+    DEPENDS
+        ${EDM4HEP_INTERFACE_INCLUDE_DIRECTORIES}/edm4hep/EDM4hepVersion.h
+        ${EDM4EIC_INTERFACE_INCLUDE_DIRECTORIES}/edm4eic/EDM4eicVersion.h
```
Comment (veprbl) [inline src/services/io/podio/CMakeLists.txt:44]: We can get some forward compat with something like
```suggestion
        ${EDM_VERSION_INCLUDES}
```
and
```cmake
set(EDM_VERSION_INCLUDES )
if(${EDM4HEP_VERSION} VERSION_GREATER_EQUAL "0.9")
list(APPEND EDM_VERSION_INCLUDES ${EDM4HEP_INTERFACE_INCLUDE_DIRECTORIES}/edm4hep/EDM4hepVersion.h)
endif()
if(${EDM4EIC_VERSION} VERSION_GREATER_EQUAL "2.1.0")
list(APPEND EDM_VERSION_INCLUDES ${EDM4EIC_INTERFACE_INCLUDE_DIRECTORIES}/edm4eic/EDM4eicVersion.h)
endif()
```

Comment (wdconinc) [reply src/services/io/podio/CMakeLists.txt:44]: or does cmake complain about that?

Comment (veprbl) [reply src/services/io/podio/CMakeLists.txt:44]: Note commented APPENDs

Comment (wdconinc) [reply src/services/io/podio/CMakeLists.txt:44]: No issues with EDM4hep < 0.9 and EDM4eic < 2.1? That is the corner case that leads to an empty DEPENDS, right.

Comment (veprbl) [reply src/services/io/podio/CMakeLists.txt:44]: The logic of my test is that, if you comment out appends, it executes as if includes are not available.

Comment (wdconinc) [reply src/services/io/podio/CMakeLists.txt:44]: Right, I see the comment characters now. Narrow mobile interface not always the best...

**Review by veprbl — APPROVED** [review]
Builds with EICrecon 2.0.0 and 2.1.0
